### PR TITLE
Added parallel axis methods for Particle and RigidBody

### PR DESCRIPTION
**Pull request by moorepants**

### CODE REVIEW

**Comment by moorepants** [comment]
@sympy/mechanics This needs a quick review.

**Comment by oliverlee** [comment]
I think this is used a few times in the code. Can you search and see if there are any instances and replace them? I'm on my phone so it's not too easy for me to do now.

File: sympy/physics/mechanics/tests/test_particle.py [hunk 1]
```diff
@@ -1,5 +1,5 @@
-from sympy import symbols
-from sympy.physics.mechanics import Point, Particle, ReferenceFrame
+from sympy import symbols, sqrt
```
Comment (oliverlee) [inline sympy/physics/mechanics/tests/test_particle.py]: why is sqrt imported

File: sympy/physics/mechanics/rigidbody.py [hunk 2]
```diff
@@ -376,3 +376,26 @@ def set_potential_energy(self, scalar):
...
+        Returns
+        =======
+        inertia : sympy.physics.vector.Dyadic
+            The inertia dyadic of the rigid body expressed about the provided
+            point.
+
+        """
+        # circular import issue
+        from sympy.physics.mechanics.functions import inertia
+        a, b, c = self.masscenter.pos_from(point).to_matrix(self.frame)
```
Comment (oliverlee) [inline sympy/physics/mechanics/rigidbody.py]: Any reason not to use xyz?

Comment (moorepants) [inline sympy/physics/mechanics/rigidbody.py]: a, b, c was in the book I looked at. I don't think it matters.

File: sympy/physics/mechanics/tests/test_particle.py (hunk 1 above)
Comment (moorepants) [inline sympy/physics/mechanics/tests/test_particle.py]: Imported and forgot to remove.

**Comment by moorepants** [comment]
There is this function https://github.com/sympy/sympy/blob/master/sympy/physics/mechanics/functions.py#L98 that is like the particle version and it is used in RigidBody to compute the central inertia. Maybe I should be using that function for this.

**Comment by moorepants** [comment]
I could have written it like this https://github.com/sympy/sympy/blob/master/sympy/physics/mechanics/rigidbody.py#L163 I suppose.

**Comment by moorepants** [comment]
I don't see any uses of the parallel axis theorem anywhere else.

**Comment by oliverlee** [comment]
Maybe I'm wrong. Thought it was implemented in rigid body because there was some bug with central inertia or inertia about a point for the rigid body class.

It doesn't matter much for abc but we use Ix Iy Iz on the documentation and it might be better for consistency. But I don't care too much.

**Comment by moorepants** [comment]
Yes, the central inertia property has this, I linked it above. But this function relies on the central inertia being available so I can't use this function to compute the central inertia.

**Comment by moorepants** [comment]
@isuruf I'm getting a symengine failure here due to a missing method. Maybe the symengine tests shouldn't be a required test on Travis. Seems like this could often be an problem to deal with that will require a new release of symengine.

**Comment by isuruf** [comment]
I agree. symengine tests should be made optional. (Btw, for a change, there doesn't need to be a release, I can upload conda packages with dev tags using https://github.com/symengine/python-symengine-feedstock)

**Comment by moorepants** [comment]
I could move the symengine test to optional here I suppose.

I understand that you can test of dev "releases" or even the latest master of symengine, but I'm not sure how good an idea that is. Seems like you'd want to test based on a stable API from dependencies, otherwise you may always be fighting test failures due to the instability.

**Comment by isuruf** [comment]
I know, but it's better this way. This failure is due to a symengine bug, but if I let this fail the tests since it's optional now, there might be more errors in the future that I won't even see. I'll fix this in symengine.py and make sure this optional test pass from time to time.

**Comment by moorepants** [comment]
So should I have the symengine test in allowed_failures or not?

**Comment by isuruf** [comment]
You can keep it optional. That's fine.

**Comment by moorepants** [comment]
This is ready to be merged @sympy/mechanics.

**Comment by czgdp1807** [comment]
@moorepants Can this still be merged? If yes, then I can continue this PR in a new one. Thanks.

**Comment by oscarbenjamin** [comment]
Why does this PR combine testing symengine on Travis with adding methods in the physics module?

If the changes to Travis were undone then there would be no merge conflict for the changes to the physics module. I think that an issue should be opened to discuss adding symengine tests if that's what is wanted. Then the other parts might be mergeable. There is no explanation of the purpose of the changes here so I'm not sure.

**Comment by moorepants** [comment]
The reason is because we were testing symengine on the sympy repo in the past. When I introduced this PR symengine was failing for an unrelated reason. So I was attempting to disable the symengine tests on the sympy repo (or not make them required). PR was left in this state. The can be updated wrt master and all would probably be fine.

**Comment by moorepants** [comment]
@czgdp1807 If you want to work on this it can be tidied up now. Maybe opening a new PR is a good idea so the release notes things will work.

**Comment by czgdp1807** [comment]
Okay, I will take this up and will create a new PR. Thanks.

**Pull request closed without merging**